openslides/agenda/models.py

```diff
@@ -133,22 +154,31 @@ def number_all(self, numeral_system='arabic'):
         """
         def walk_tree(tree, number=None):
             for index, tree_element in enumerate(tree):
+                # Calculate number of visable agenda items.
                 if numeral_system == 'roman' and number is None:
                     item_number = to_roman(index + 1)
                 else:
                     item_number = str(index + 1)
                     if number is not None:
                         item_number = '.'.join((number, item_number))
+                # Add prefix.
                 if config['agenda_number_prefix']:
                     item_number_tmp = "%s %s" % (config['agenda_number_prefix'], item_number)
                 else:
                     item_number_tmp = item_number
+                # Save the new value and go down the tree.
                 tree_element['item'].item_number = item_number_tmp
                 tree_element['item'].save()
                 walk_tree(tree_element['children'], item_number)
 
+        # Start numbering visable agenda items.
         walk_tree(self.get_tree(only_agenda_items=True, include_content=True))
 
+        # Reset number of hidden items.
+        for item in self.get_only_hidden_items():
+            item.item_number = ''
+            item.save()
```

Comment:
Proposal:
This is a case where you could work with update()
```python
from openslides.utils.autoupdate import inform_changed_data
hidden_items = self.get_only_hidden_items()  # todo, rename method to get_hidden_items :)
Item.objects.filter(pk__in=[item.pk for item in hidden_items]).update(item_number='')
inform_changed_data(*hidden_items)
```
This is much more efficient if there are a lot of hidden items.